addons/project/views/project_views.xml

```diff
@@ -1709,8 +1709,16 @@
             <field name="view_id" ref="view_task_all_calendar"/>
         </record>
 
+        <record id="action_menu_view_all_task" model="ir.actions.server">
+            <field name="name">menu View all task</field>
+            <field name="model_id" ref="project.model_project_task"/>
+            <field name="state">code</field>
+            <field name="code">
+                action = model.action_menu_view_all_task()
```

Comment:
I am not a big fan to create something in sudo in a public method because a rpc could call this method and potentially flood the db. 😬 

We could instead, convert the method in private and just ensure the current has his personal stage.

Something like this:
```suggestion
                model._ensure_personal_stages_exist()
                action = model.env['ir.actions.act_window']._for_xml_id('project.action_view_all_task')
```

(I am not sure the last line work.) 🙂